eqcorrscan/core/match_filter.py

```diff
@@ -3490,26 +3501,27 @@ def normxcorr2(template, image):
         correlation of the image with the template.
     :rtype: numpy.ndarray
     """
-    from eqcorrscan.utils.correlate import fftw_normxcorr
+    array_xcorr = get_array_xcorr()
     # Check that we have been passed numpy arrays
     if type(template) != np.ndarray or type(image) != np.ndarray:
         print('You have not provided numpy arrays, I will not convert them')
         return 'NaN'
     if len(template) > len(image):
-        ccc = fftw_normxcorr(
+        ccc = array_xcorr(
             templates=np.array([image]).astype(np.float32),
             stream=template.astype(np.float32), pads=[0],
             threaded=False)[0][0]
     else:
-        ccc = fftw_normxcorr(
+        ccc = array_xcorr(
             templates=np.array([template]).astype(np.float32),
             stream=image.astype(np.float32), pads=[0], threaded=False)[0][0]
     ccc = ccc.reshape((1, len(ccc)))
     return ccc
 
 
 def match_filter(template_names, template_list, st, threshold,
-                 threshold_type, trig_int, plotvar, plotdir='.', cores=None,
+                 threshold_type, trig_int, plotvar, plotdir='.',
+                 xcorr_func=None, concurrency=None, cores=None,
```

Comment:
I think these extra options probably need to be included in:
- _group_detect (both below methods call this function)
- Template.detect
- Tribe.detect

Reply:
Ok, should the parallel_process option be removed in favor of the concurrency argument? Maybe it should just be depreciated? If so do we have an official way of depreciating things?

Reply:
So `parallel_process` is just supposed to determine whether the data should be pre-processed in parallel, it *should* be separate to the actual matched-filtering, so, we should be able to leave that, and just add in the `concurrency` and `xcorr_func` kwargs?